src/transformers/training_args.py

```diff
@@ -217,6 +217,10 @@ class TrainingArguments:
         sharded_ddp (:obj:`bool`, `optional`, defaults to :obj:`False`):
             Use Sharded DDP training from `FairScale <https://github.com/facebookresearch/fairscale>`__ (in distributed
             training only). This is an experimental feature.
+        deepspeed (:obj:`bool`, `optional`, defaults to :obj:`False`):
+            Use `Deepspeed <https://github.com/microsoft/deepspeed>`__. This is an experimental feature.
+        deepspeed_config (:obj:`str`, `optional`):
```

Comment:
```suggestion
        deepspeed_config_file (:obj:`str`, `optional`):
```
This makes it clearer to me we should pass along a file and not a config object (which the library has plenty of).

Reply:
`deepspeed.initialize` expects to find `args.deepspeed_config` so if we follow your suggestion we will have to rewrite that key before passing `args` to `deepspeed.initialize`.

As I mentioned elsewhere I think it'd be sufficient to just have a single argument `deepspeed` and have its value to be the config file and then re-assign it to `args.deepspeed_config` before `deepspeed.initialize` .

But either your suggestion or mine will break the deepspeed convention of:
```
deepspeed myprog myargs --deepspeed --deepspeed_config ds_config.js
```
so it'd be slightly confusing to users.

Reply:
Oh I didn't realize there is a convention on their side. So let's not break it then, and keep the args as you did. A mention in bold of the doc that this should be a config file will be necessary.

Reply:
Looks like `args.local_rank` is needed too, please have a look at how I made it clear what's being passed to deepspeed:
https://github.com/huggingface/transformers/pull/9211/commits/9cc3b63d5a4049a3660c23eed96c8456b3457b10

Reply:
So since I'm rewrapping them anyway - it's your call now - I can generate the 2 vars on the fly based on just `deepspeed_config_file` you suggested - though it is kind of an odd name for double function. From user's point of view If I pass a config file, does it also activate the feature? I suppose this is why they had 2 vars. Not sure.

@patrickvonplaten, what do you think? 

See my comment here https://github.com/huggingface/transformers/pull/9211#discussion_r547031679
and then we are considering to collapse these 2 into a single cl arg that will provide both - the config file and at the same time activate deepspeed.

Reply:
I also asked for suggestions at https://github.com/microsoft/DeepSpeed/issues/616

Reply:
So I got feedback and no, there is no need to use both vars, in fact `--deepspeed` is not needed at all.

So let's use a single cl arg.

Let's just decide how to best name it.

So the proposals so far:

1. I propose `--deepspeed ds_config.js`. While it's less obvious that it expects a file name argument, it's unambiguous about it activating deepspeed. 
2. We could make the value even optional and default to `ds_config.js`, so most of the time it'd be just `--deepspeed`
3. Your proposal @sgugger was `--deepspeed_config_file` but that was in combination with `--deepspeed`

What are your thoughts?

Reply:
Well, until everybody is back I changed it to just one cl arg: `--deepspeed ds_config.js`. If you prefer a different name please let me know - should be easy to rename.